src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs

```diff
@@ -132,11 +133,25 @@ public static partial class JsonSerializer
                         clearMax = bytesInBuffer;
                     }
 
+                    int start = 0;
+                    if (loopCount == 0)
+                    {
+                        // Handle the UTF-8 BOM if present
+                        int utf8BomLength = JsonConstants.Utf8Bom.Length;
+                        if (buffer.Length >= utf8BomLength && JsonConstants.Utf8Bom.SequenceEqual(buffer.AsSpan(0, utf8BomLength).Slice(0, utf8BomLength)))
```

Comment:
I am wondering what happens if `JsonSerializerOptions.DefaultBufferSize` was set to 1 (or even 2), in which case `buffer.Length == 1`. Then, we wouldn't end up ignoring the BOM in the first iteration, and never hit this code path again.

Reply:
There is no guarantee that the buckets for the `ArrayPool` won't change in the future. You can always add validation tests to see that the rented size is what you expect if you are depending on it.

Reply:
One option would be to rent `Math.Max(options.DefaultBufferSize, JsonConstants.Utf8Bom.Length)`; now the array is always guaranteed to have `length >= JsonConstants.Utf8Bom.Length`

Reply:
> One option would be to rent Math.Max(options.DefaultBufferSize, JsonConstants.Utf8Bom.Length); now the array is always guaranteed to have length >= JsonConstants.Utf8Bom.Length

That would mean DefaultBufferSize will not really be respected when it is < 3. I'm ok with that, because the fix is straightforward. @ahsonkhan

Reply:
Given we already read 16 at a minimum currently due to the array pool behavior (and hence don't follow the `DefaultBufferSize` precisely), I am ok with this. I am also OK with it being bounded at the bottom by 16. We could even consider throwing for anything below (rather than throwing for < 1), but maybe that's not necessary. cc @steveharter 

Even if someone says `DefaultBufferSize = 10k`, we still end up rounding up to the nearest power of 2 (or I should say, the underlying array pool does).